### DISCUSSION TITLE
className support

### DISCUSSION THREAD

**Discussion by Chirag-kalsariya** [jaredh159/tailwind-react-native-classnames#304] [General]
currently, we can style native components using syntax like this.

```
<View style={tw`p-4 android:pt-2 bg-white dark:bg-black`}>
    <Text style={tw`text-md text-black dark:text-white`}>Hello World</Text>
 </View>
```

What about support of className for styling. 

```
<View className="p-4 android:pt-2 bg-white dark:bg-black">
    <Text  className="text-md text-black dark:text-white">Hello World</Text>
 </View>
```

The reason for that is we can use other library like class-variance-authority, and clsx. Using this we can create components variants. And, it will benefits to create component library like Shadcn for react-native.

I already try Nativewind, and Nativecn, but they are too unstable. also, not supporting Monorepo, and complex to setup for web.

**Comment by jaredh159** [comment]
he there, thanks for opening the discussion. it's a good question. my current stance on this is pretty well summed up by my comment on another issue: https://github.com/jaredh159/tailwind-react-native-classnames/issues/181#issuecomment-1201183852
